app/services/licence-monitoring-station/setup/submit-full-condition.service.js

```diff
@@ -0,0 +1,95 @@
+'use strict'
+
+/**
+ * Orchestrates validating the data for `/licence-monitoring-station/setup/{sessionId}/full-condition`
+ *
+ * @module SubmitFullConditionService
+ */
+
+const FullConditionService = require('../../../services/licence-monitoring-station/setup/full-condition.service.js')
+const FullConditionValidator = require('../../../validators/licence-monitoring-station/setup/full-condition.validator.js')
+const LicenceVersionPurposeConditionModel = require('../../../models/licence-version-purpose-condition.model.js')
+const SessionModel = require('../../../models/session.model.js')
+
+/**
+ * Orchestrates validating the data for `/licence-monitoring-station/setup/{sessionId}/full-condition`
+ *
+ * @param {string} sessionId
+ * @param {object} payload - The submitted form data
+ *
+ * @returns {Promise<object>} - The data formatted for the view template
+ */
+async function go(sessionId, payload) {
+  const validationResult = _validate(payload)
+
+  if (!validationResult) {
+    const session = await SessionModel.query().findById(sessionId)
+
+    const conditionAbstractionPeriod = await _fetchAbstractionPeriod(payload.condition)
+
+    await _save(session, conditionAbstractionPeriod, payload)
+
+    // If the user selected a non-condition option then they will proceed to the "enter abstraction period" page.
+    // Ordinarily we would also return `checkPageVisited` to say whether the user should be forwarded there; however,
+    // if the user has selected a condition option then they've reached the end of the journey so will go to the check
+    // page, and if they selected a non-condition option then they must enter an abstraction period no matter what.
+    // We therefore don't care about `checkPageVisited` on this particular page.
+    return {
+      abstractionPeriod: payload.condition === 'not_listed'
+    }
+  }
+
+  const pageData = await FullConditionService.go(sessionId)
+
+  return {
+    error: validationResult,
+    ...pageData
+  }
+}
+
+async function _fetchAbstractionPeriod(conditionId) {
+  if (conditionId === 'not_listed') {
+    return null
+  }
+
+  return LicenceVersionPurposeConditionModel.query()
```

Comment:
Do we need to worry about extracting the abstraction period from the linked condition? From looking at existing data in the `licence_monitoring_stations` view it appears that if linked to a condition then the abstraction period is not saved in this table. So there should be no need to add it to the session when linked to a condition?